From 35ced1311d7faaa9bab8318e35f846a6a576e822 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Wed, 25 Mar 2015 18:14:40 +0800
Subject: [PATCH] MLK-10461: arm: imx7: set the PFD override bit before system
 entering low power mode

the PFD override bit must be set before system entering any low power mode.

Signed-off-by: Bai Ping <b51503@freescale.com>
---
 arch/arm/mach-imx/anatop.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/arch/arm/mach-imx/anatop.c b/arch/arm/mach-imx/anatop.c
index d37b579..14535f6 100644
--- a/arch/arm/mach-imx/anatop.c
+++ b/arch/arm/mach-imx/anatop.c
@@ -100,10 +100,10 @@ static void imx_anatop_disable_pu(bool off)
 void imx_anatop_pre_suspend(void)
 {
 	if (cpu_is_imx7d()) {
-		/* PLL overwrite set */
+		/* PLL and PFDs overwrite set */
 		regmap_write(anatop, ANADIG_ARM_PLL + REG_SET, 1 << 20);
 		regmap_write(anatop, ANADIG_DDR_PLL + REG_SET, 1 << 19);
-		regmap_write(anatop, ANADIG_SYS_PLL + REG_SET, 1 << 17);
+		regmap_write(anatop, ANADIG_SYS_PLL + REG_SET, 0x1ff << 17);
 		regmap_write(anatop, ANADIG_ENET_PLL + REG_SET, 1 << 13);
 		regmap_write(anatop, ANADIG_AUDIO_PLL + REG_SET, 1 << 24);
 		regmap_write(anatop, ANADIG_VIDEO_PLL + REG_SET, 1 << 24);
@@ -127,10 +127,10 @@ void imx_anatop_pre_suspend(void)
 void imx_anatop_post_resume(void)
 {
 	if (cpu_is_imx7d()) {
-		/* PLL overwrite clear */
+		/* PLL and PFDs overwrite clear */
 		regmap_write(anatop, ANADIG_ARM_PLL + REG_CLR, 1 << 20);
 		regmap_write(anatop, ANADIG_DDR_PLL + REG_CLR, 1 << 19);
-		regmap_write(anatop, ANADIG_SYS_PLL + REG_CLR, 1 << 17);
+		regmap_write(anatop, ANADIG_SYS_PLL + REG_CLR, 0x1ff << 17);
 		regmap_write(anatop, ANADIG_ENET_PLL + REG_CLR, 1 << 13);
 		regmap_write(anatop, ANADIG_AUDIO_PLL + REG_CLR, 1 << 24);
 		regmap_write(anatop, ANADIG_VIDEO_PLL + REG_CLR, 1 << 24);
-- 
1.8.0

